bundles/org.openhab.binding.openwebnet/src/main/java/org/openhab/binding/openwebnet/handler/OpenWebNetEnergyHandler.java

```diff
@@ -49,12 +56,95 @@
+        if (h != null && h.isBusGateway()) {
+            OpenGateway gw = h.gateway;
+            if (gw != null && gw.isConnected()) {
+                // bridge is online
+                subscribeToActivePowerChanges();
+            }
+        }
+    }
+
+    @Override
+    public void bridgeStatusChanged(ThingStatusInfo bridgeStatusInfo) {
+        super.bridgeStatusChanged(bridgeStatusInfo);
+
+        // subscribe the scheduler only after the bridge is online
+        if (bridgeStatusInfo.getStatus().equals(ThingStatus.ONLINE)) {
+            subscribeToActivePowerChanges();
+        }
+    }
+
+    private void subscribeToActivePowerChanges() {
+        notificationSchedule = scheduler.scheduleAtFixedRate(() -> {
+            if (isFirstSchedulerLaunch) {
+                logger.debug(
+                        "subscribeToActivePowerChanges() For WHERE={} subscribing to active power changes notification for the next {}min",
+                        deviceWhere, ENERGY_SUBSCRIPTION_PERIOD);
+            } else {
+                logger.debug(
+                        "subscribeToActivePowerChanges() Refreshing subscription for the next {}min for WHERE={} to active power changes notification",
+                        ENERGY_SUBSCRIPTION_PERIOD, deviceWhere);
+            }
+
+            try {
+                bridgeHandler.gateway.send(EnergyManagement.setActivePowerNotificationsTime(deviceWhere.value(),
+                        ENERGY_SUBSCRIPTION_PERIOD));
+                isFirstSchedulerLaunch = false;
+            } catch (Exception e) {
+                if (isFirstSchedulerLaunch) {
+                    logger.warn(
+                            "subscribeToActivePowerChanges() For WHERE={} could not subscribe to active power changes notifications. Exception={}",
+                            deviceWhere, e.getMessage());
+                } else {
+                    logger.warn(
+                            "subscribeToActivePowerChanges() Unable to refresh subscription to active power changes notifications for WHERE={}. Exception={}",
+                            deviceWhere, e.getMessage());
+                }
+            }
+        }, 0, ENERGY_SUBSCRIPTION_PERIOD - 1, TimeUnit.MINUTES);
+    }
+
+    @Override
+    public void dispose() {
+        if (notificationSchedule != null) {
+            notificationSchedule.cancel(false);
+        }
+        super.dispose();
+        scheduler.schedule(() -> {
+            try {
+                // switch off active power updates
+                bridgeHandler.gateway.send(EnergyManagement.setActivePowerNotificationsTime(deviceWhere.value(), 0));
+            } catch (Exception e) {
+                logger.warn(
+                        "dispose() For WHERE={} could not UN-subscribe from active power changes notifications. Exception={}",
+                        deviceWhere, e.getMessage());
+            }
+        }, 1, TimeUnit.MINUTES);
```

Comment:
After a Thing or Bridge has been disposed, no tasks shall be run. Also, `dispose()` should return fast, so network activtiy should be done in this method only if necessary. Can you elaborate what `setActivePowerNotificationsTime()` is doing here?

Reply:
Understood. What happens if you don't send this message in `dispose()`?

Reply:
the probe continues to send data over the bus until expiration (10 minutes in the worst case)

Reply:
Doesn't sound too bad. Why not remove it from `dispose()` completely?

Reply:
leaving the subscription active and expire by itself is not a big issue at all. 
But what if in other cases there is a operation involving network messages that must be performed when a Thing is removed ? Where one should do this if not in the dispose() ? (just in case, for future reference)

Reply:
There is at least one binding that requires network access in `dispose()` (velux). But that is caused by a very bad implementation by Velux in their product.